src/figuresClasses.ts

```diff
@@ -1,11 +1,85 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
```

Comment:
[CODE KNOWLEDGE] - You do not need to assign the constructor parameters to class properties manually if you are using access modifiers like 'public' or 'private'. TypeScript does this automatically. You can simplify the constructor by removing the property assignments.

GOOD EXAMPLE:
constructor(public color: Color, public a: number, public b: number, public c: number) {
  // constructor logic
}